src/code/relocation.c

```diff
@@ -37,55 +37,60 @@ void Overlay_Relocate(void* allocatedVRamAddress, OverlayRelocationSection* over
                 }
                 break;
-            case 0x4000000:
-                /* R_MIPS_26
-                 * Handles 26-bit address relocation, used for jumps and jals
-                 */
-                unrelocatedAddress = ((*relocDataP & 0x3FFFFFF) << 2) | 0x80000000;
-                relocOffset = unrelocatedAddress - (u32)vRamAddress;
+
+            case R_MIPS_26 << 24:
+                // Handles 26-bit address relocation, used for jumps and jals.
+                // Extract the address from the target field of the J-type MIPS instruction.
+                // Relocate the address and update the instruction.
+
+                unrelocatedAddress = PHYS_TO_K0((*relocDataP & 0x3FFFFFF) << 2);
+                relocOffset = unrelocatedAddress - (uintptr_t)vRamStart;
                 relocatedValue = (*relocDataP & 0xFC000000) | (((allocu32 + relocOffset) & 0xFFFFFFF) >> 2);
-                relocatedAddress = ((relocatedValue & 0x3FFFFFF) << 2) | 0x80000000;
+                relocatedAddress = PHYS_TO_K0((relocatedValue & 0x3FFFFFF) << 2);
                 *relocDataP = relocatedValue;
                 break;
-            case 0x5000000:
-                /* R_MIPS_HI16
-                 * Handles relocation for a lui instruciton, store the reference to
-                 * the instruction, and will update it in the R_MIPS_LO16 section.
-                 */
+
+            case R_MIPS_HI16 << 24:
+                // Handles relocation for a hi/lo pair, part 1.
+                // Store the reference to the LUI instruction (hi) using the `rt` register of the instruction.
+                // This will be updated later in the `R_MIPS_LO16` section.
+
                 luiRefs[(*relocDataP >> 0x10) & 0x1F] = relocDataP;
                 luiVals[(*relocDataP >> 0x10) & 0x1F] = *relocDataP;
                 break;
-            case 0x6000000:
-                /* R_MIPS_LO16
-                 * Updates the LUI instruction to reflect the relocated address.
-                 * The full address is calculated from the LUI and lo parts, and then updated.
-                 * if the lo part is negative, add 1 to the lui.
-                 */
-                regValP = &luiVals[((*relocDataP >> 0x15) & 0x1F)];
-                vaddr = (*regValP << 0x10) + (s16)*relocDataP;
-                luiInstRef = luiRefs[((*relocDataP >> 0x15) & 0x1F)];
-                if ((vaddr & 0xF000000) == 0) {
-                    relocOffset = vaddr - (u32)vRamAddress;
-                    vaddr = (s16)relocData;
+
+            case R_MIPS_LO16 << 24:
+                // Handles relocation for a hi/lo pair, part 2.
+                // Grab the stored LUI (hi) from the `R_MIPS_HI16` section using the `rs` register of the instruction.
+                // The full address is calculated, relocated, and then used to update both the LUI and lo instructions.
+                // If the lo part is negative, add 1 to the LUI.
```

Comment:
```suggestion
                // If the lo part is negative, add 1 to the hi value.
```
or something, this could be read as "changing the opcode"